lib/puppet/indirector/request.rb

```diff
@@ -42,7 +42,8 @@ def environment=(env)
   end
 
   def escaped_key
-    URI.escape(key)
+    # TODO: these keys are generally fragments so Puppet::Util.uri_escape isn't really appropriate
+    Puppet::Util.uri_escape(key)
   end
```

Comment:
I think the key is mostly (always?) part of the path and not the query so if I'm understanding `uri_escape` correctly this is the right thing to do.